### ISSUE TITLE
extended test builds never getting scheduled

### ISSUE THREAD

**Issue by gabemontero** [openshift/origin#14544]
I've seen this about 5 times now ... if there was an issue already opened I've missed it.

Random builds (of varying strategy types) in our extended builds do not get scheduled.  I see this type of event in the event dump:

```
Jun  8 18:51:19.181: INFO: jenkins-sync-plugin-snapshot-test-1-build  172.18.11.208  Pending         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:51:05 -0400 EDT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:51:05 -0400 EDT ContainersNotReady containers with unready status: [docker-build]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:51:05 -0400 EDT  }]
``` 

The latest incarnation was https://ci.openshift.redhat.com/jenkins/job/test-openshift-sync-plugin/37/consoleFull

I've marked it component/build for now, but am not certain that is the right place.

@openshift/devex

**Comment by bparees** [comment]
hm, the failure I see in that log is that jenkins itself failed to deploy:

```
Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:50 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} Pulling: pulling image "openshift/origin-deployer:6acabdc"
Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:50 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "deployment" with ErrImagePull: "rpc error: code = 2 desc = Tag 6acabdc not found in repository docker.io/openshift/origin-deployer"

Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:50 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} Failed: Failed to pull image "openshift/origin-deployer:6acabdc": rpc error: code = 2 desc = Tag 6acabdc not found in repository docker.io/openshift/origin-deployer
Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:51 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "deployment" with ImagePullBackOff: "Back-off pulling image \"openshift/origin-deployer:6acabdc\""

Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:51 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} BackOff: Back-off pulling image "openshift/origin-deployer:6acabdc"
```

the jenkins sync plugin snapshot test build appears to have succeeded:
```
Jun  8 17:50:48.467: INFO: jenkins-sync-plugin-snapshot-test-1-build  172.18.11.208  Succeeded         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:34:16 -0400 EDT PodCompleted } {Ready False 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:35:46 -0400 EDT PodCompleted } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:34:16 -0400 EDT  }]
```

So the question is why we're trying to use a deploy image that apparently isn't available locally or in docker hub.  @stevekuznetsov @mfojtik

**Comment by bparees** [comment]
ok I see the failure @gabemontero was looking at also, but it has the same failure reason:

```
Jun  8 18:51:19.177: INFO: At 2017-06-08 17:51:09 -0400 EDT - event for jenkins-sync-plugin-snapshot-test-1-build: {kubelet 172.18.11.208} BackOff: Back-off pulling image "openshift/origin-docker-builder:6acabdc"
Jun  8 18:51:19.177: INFO: At 2017-06-08 17:51:
```

ultimately i think this is another case of image GC deleting the deployer image out from under us.

**Comment by gabemontero** [comment]
On Fri, Jun 9, 2017 at 12:35 PM, Ben Parees <notifications@github.com>
wrote:

> hm, the failure I see in that log is that jenkins itself failed to deploy:
>
> Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:50 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} Pulling: pulling image "openshift/origin-deployer:6acabdc"
> Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:50 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "deployment" with ErrImagePull: "rpc error: code = 2 desc = Tag 6acabdc not found in repository docker.io/openshift/origin-deployer"
>
> Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:50 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} Failed: Failed to pull image "openshift/origin-deployer:6acabdc": rpc error: code = 2 desc = Tag 6acabdc not found in repository docker.io/openshift/origin-deployer
> Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:51 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "deployment" with ImagePullBackOff: "Back-off pulling image \"openshift/origin-deployer:6acabdc\""
>
> Jun  8 17:50:48.463: INFO: At 2017-06-08 17:35:51 -0400 EDT - event for jenkins-1-deploy: {kubelet 172.18.11.208} BackOff: Back-off pulling image "openshift/origin-deployer:6acabdc"
>
> the jenkins sync plugin snapshot test build appears to have succeeded:
>
> Jun  8 17:50:48.467: INFO: jenkins-sync-plugin-snapshot-test-1-build  172.18.11.208  Succeeded         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:34:16 -0400 EDT PodCompleted } {Ready False 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:35:46 -0400 EDT PodCompleted } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-06-08 17:34:16 -0400 EDT  }]
>
> So the question is why we're trying to use a deploy image that apparently
> isn't available locally or in docker hub. @stevekuznetsov
> <https://github.com/stevekuznetsov> @mfojtik <https://github.com/mfojtik>
>
There are 2 failures in that PR test run.  The failed deploy you noted, and
then the  build I noted.  This bug was opened for the latter.

**Comment by stevekuznetsov** [comment]
Can we get that job using `oct` and the `sjb/` generators? Should be a fairly simple switch. I'm debugging a similar image GC issue with Logging right now -- I think something's screwed up with the Vagrant provisioning but I just don't care to figure out why. Would much rather move to the new jobs and not look back.

**Comment by bparees** [comment]
@stevekuznetsov i assume your point is if we switch to that we'll be able to easily allocate more storage?

**Comment by gabemontero** [comment]
On Fri, Jun 9, 2017 at 1:10 PM, Ben Parees <notifications@github.com> wrote:

> Can we get that job using oct and the sjb/ generators? Should be a fairly
> simple switch.
>
> we'll queue that up for post 3.6 feature cut.
>
Just told @stevekuznetsov the same thing on IRC a couple of minutes ago :-)

**Comment by stevekuznetsov** [comment]
Not more storage, that's simple enough to do with Vagrant ... more interested in all of the investments we've made on logging, metrics, artifacts ... if this were a new job we could in ten seconds look at the filesystem info we gather and know for certain how much disk pressure there was.

**Comment by bparees** [comment]
@gabemontero i'll let you handle the job migration on this one.

**Comment by gabemontero** [comment]
I've started this process.  `oct` has been updated to recognize the various plugin repos.  A base ami update is cooking as I type.  I have a prototype of the aos-cd-jobs change on my laptop.  Once the AMI's have the new `oct`, I plan to leverage the instructions at https://github.com/openshift/aos-cd-jobs#custom-xml-generator-configuration-under-sjb in order to test the associated new jobs.

**Comment by gabemontero** [comment]
Quick update ... iterating on  https://github.com/openshift/ci-config/pull/8  ... pending approval and push to ci.openshift to update plugin jobs

**Comment by gabemontero** [comment]
Conversion to the new infa is in place for the 4 jenkins plugin extended tests; during final verifications I ran both the vagrant based and oct based jobs in parallel.  Saw several times that the vagrant job would fail with odd-ball, env suspect reasons, and the oct job was clean.

Closing this out.  If something comes up with the oct based jobs a separate issue is warranted.
